### DISCUSSION TITLE
[Support]: what does loitering_time in 0.14 actually mean?

### DISCUSSION THREAD

**Discussion by d-steve** [blakeblackshear/frigate#12204] [General Support] [answered]
### Describe the problem you are having

Loitering_time seems to be defined as the time loitering without defining exactly what loitering is. 

So, I can’t tell what the difference is between loitering_time and inertia other than expressed as seconds vs frames. 

What is loitering, and how is this different than inertia? Thanks!

### Version

0.14-beta3

### Frigate config file

```yaml
NA
```

### Relevant log output

```shell
NA
```

### FFprobe output from your camera

```shell
NA
```

### Operating system

Debian

### Install method

Docker Compose

### Object Detector

Coral

### Network connection

Wired

### Camera make and model

Dahua

**Comment by hawkeye217** [comment]
From the docs: https://deploy-preview-11419--frigate-docs.netlify.app/configuration/zones#loitering-time

> Zones support a loitering_time configuration which can be used to only consider an object as part of a zone if they loiter in the zone for the specified number of seconds. This can be used, for example, to create alerts for cars that stop on the street but not cars that just drive past your camera.

**Reply by d-steve** [reply]
Yeah, I read that but don’t know what “loiter” means. I would interpret it as the same as inertia but expressed in seconds instead of frames. 

Does “loiter” mean it has to be stopped but “inertia” means in zone regardless of moving or stopped?

**Reply by hawkeye217** [reply]
Have a look at the code:

https://github.com/blakeblackshear/frigate/blob/53a2a865f112f622b56e842503dd101a8489e1de/frigate/object_processing.py#L205

Inertia will determine how many frames it takes for an object to be considered part of a zone. Once the inertia threshold has been met, loitering time will determine when the object is actually marked as part of the zone. The default loitering time is 0, so after the inertia condition has been met, the object is marked as part of the zone. This is delayed by `loitering_time` seconds, if set.

**Reply by d-steve** [reply]
Ah, thx. I can’t tell a meaningful difference between inertia and loitering. 

So loitering and inertia appear to both be simply delaying when an object is marked in zone, and loitering score doesn’t increment until after inertia is met.

That is, is there a real world difference at 5fps between, say, inertia=23, loitering_time=0 vs inertia=3, loitering_time=4 ?

**Reply by NickM-27** [reply, marked as answer]
in 0.13 inertia could technically be used for the same thing, but they are really designed for different use cases. In general inertia should not be messed with and loitering should be used
